soc/xtensa/intel_apl_adsp/include/platform/io.h

```diff
@@ -0,0 +1,53 @@
+/* SPDX-License-Identifier: BSD-3-Clause
```

Comment:
Zephyr has `sys_read16()`, `sys_write32()`, etc. So there is no need to define these, and can simply be alias to `sys_*` functions.

Reply:
Yes, this would be better, I just copied the header from SOF

Reply:
I have some error when sourcing this from sys_io.h
```
In file included from /home/niko/work/sof/zephyrproject/zephyr/include/toolchain.h:39,
                 from /home/niko/work/sof/zephyrproject/zephyr/include/arch/common/sys_io.h:17,
                 from /home/niko/work/sof/zephyrproject/zephyr/soc/xtensa/intel_apl_adsp/include/platform/io.h:15,
                 from /home/niko/work/sof/zephyrproject/zephyr/soc/xtensa/intel_apl_adsp/include/platform/clk.h:13,
                 from /home/niko/work/sof/zephyrproject/zephyr/soc/xtensa/intel_apl_adsp/include/platform/platform.h:30,
                 from /home/niko/work/sof/zephyrproject/zephyr/boards/xtensa/up_squared_adsp/bootloader/boot_loader.c:10:
/home/niko/work/sof/zephyrproject/zephyr/include/toolchain/gcc.h:413:2: error: #error processor architecture not supported
  413 | #error processor architecture not supported
      |  ^~~~~
```

Reply:
I will not use this for now, see comment https://github.com/zephyrproject-rtos/zephyr/pull/20221#issuecomment-549727313

Reply:
The issue was with compile options when building bootloader code, since we do not include cmake app tempaltes, fixed now.